model/src/main/java/com/javahelp/model/token/Token.java

```diff
@@ -0,0 +1,123 @@
+package com.javahelp.model.token;
+
+import com.javahelp.model.user.User;
+import org.joda.time.DateTime;
+import java.security.SecureRandom;
+
+/**
+ * Representation of token entity
+ */
+
+public class Token {
+    /**
+     * Token requested by the user
+     */
+    private String token;
+
+    /**
+     * Unique id of the token.
+     */
+    private String id;
+
+    /**
+     * Expiry date of the token.
+     */
+    private DateTime expiry;
+
+    /**
+     * Issued date of the token.
+     */
+    private DateTime issued;
+
+    /**
+     * User who requested the token.
+     */
+    private User user;
+
+    /**
+     * This holds any additional information about the toekn.
+     */
+    private String tag;
+
+
+    /**
+     * Creates a new {@link Token}
+     * @param tag {@link String} of this {@link Token}
+     * @param user {@link User} who requested this {@link Token}
+     */
+    public Token(String tag, User user){
+        // Creating a token using functions from SecureRandom
+
+        SecureRandom random = new SecureRandom();
+        byte bytes[] = new byte[20];
+        random.nextBytes(bytes);
+        this.token = bytes.toString();
+
+        this.issued = new DateTime(DateTime.now());
+        this.expiry = this.issued.plusDays(365/2);
+        this.user = user;
+        this.tag = tag;
+    }
```

Comment:
We need a way to assign a specific token string to a `Token` object, it would be best if this was possible from the constructor. This constructor should take an additional `String` argument that sets the `Token` private field, instead of randomly generating a byte array to put in the private field.